### PR TITLE
fix: Clarify existing resources/RestAPI usage

### PR DESCRIPTION
**Pull request by nadetastic**
#### Description of changes:

- Clarify the usage of Amplify.getConfig() one Rest API/using existing resources section.
- Updated JS platforms for `InlineFilter` usage

#### Related GitHub issue #, if available:
https://github.com/aws-amplify/amplify-js/issues/12696
https://github.com/aws-amplify/amplify-js/issues/12685

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/restapi/existing-resources/index.mdx [hunk 1]
```diff
@@ -42,18 +42,38 @@ import flutter2 from '/src/fragments/lib/restapi/existing-resources.mdx';
...
-    vue: js3
-  }}
-/>
+<InlineFilter filters={['javascript','angular','nextjs','react','vue','react-native']}>
 
-import rn4 from '/src/fragments/lib/restapi/js/existing-resources.mdx';
+Existing Amazon API Gateway resources can be used with the Amplify Libraries by calling `Amplify.configure()` with the API Gateway API name and options. Note you need to supply the full resource configuration and library options objects when calling `Amplify.configure()`. The following example shows how to configure additional API Gateway resources to an existing Amplify application:
 
-<Fragments fragments={{ 'react-native': rn4 }} />
+```javascript
```
Comment (HuiSF) [inline src/pages/[platform]/build-a-backend/restapi/existing-resources/index.mdx:50]: nit: ts

**Review by HuiSF — APPROVED** [review]

**Review by cwomack — DISMISSED** [review]

**Review by cwomack — DISMISSED** [review]

**Review by HuiSF — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/restapi/existing-resources/index.mdx [hunk 2]
```diff
@@ -43,11 +43,17 @@ import flutter2 from '/src/fragments/lib/restapi/existing-resources.mdx';
...
+Existing Amazon API Gateway resources can be used with the Amplify Libraries by calling `Amplify.configure()` with the API Gateway API name and options. Note, you need to supply the full resource configuration and library options objects when calling `Amplify.configure()`. The following example shows how to configure additional API Gateway resources to an existing Amplify application:
+
+```typescript
+import { Amplify } from 'aws-amplify';
+import amplifyconfig from './src/amplifyconfiguration.json';
+Amplify.configure(amplifyconfig):
+
 const existingConfig = Amplify.getConfig();
 
+// Add existing resource to the existing configuration.
```
Comment (kvramyasri7) [inline src/pages/[platform]/build-a-backend/restapi/existing-resources/index.mdx:56]: nit: We can add backend / AWS / infrastructure resources
```suggestion
// Add existing resources to the existing configuration.
```

Comment (nadetastic) [reply src/pages/[platform]/build-a-backend/restapi/existing-resources/index.mdx:56]: Comment is for this example where we are adding only one resource 😊

**Review by kvramyasri7 — DISMISSED** [review]

File: yarn.lock
Comment (timngyn) [inline yarn.lock:1]: Was this changed by accident? If you didn't need to update the package versions, would it be okay to just revert this file?

**Review by timngyn — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (nadetastic) [reply yarn.lock:1]: reverted changes

**Review by cwomack — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by timngyn — DISMISSED** [review]

**Review by cwomack — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**